src/app/services/compute.service.ts

```diff
@@ -93,22 +127,53 @@ public number;
 
     public listSIG(subscriptionId: string, location: string): Observable<Resource[]> {
         const params = new HttpParams()
-            .set("$filter", `resourceType eq  and location eq '${location}'`);
+            .set("$filter", `resourceType eq '${computeGalleryImageVersionProvider}' and location eq '${location}'`);
         const options = { params };
-
-        return this.subscriptionService.get(subscriptionId).pipe(
+        const x = this.subscriptionService.get(subscriptionId).pipe(
             flatMap((subscription) => {
-                return this.azure.get<ArmListResponse>(subscription, resourceUrl(subscriptionId), options).pipe(
-                    expand(obs => {
-                        return obs.nextLink ? this.azure.get(subscription, obs.nextLink, options) : empty();
+                const locationsUri = `subscriptions/${subscription.subscriptionId}/locations`;
+                return this.azure.get<ArmListResponse<ArmLocationAttributes>>(subscription, locationsUri).pipe(
+                    flatMap(response => response.value.map(x => new ArmLocation(x))),
```

Comment:
I don't think this should be a `flatMap` but just a `map` does this work?

Reply:
yea that works

Reply:
forces the next line to be a double map as well though.